controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,162 @@
+package storagecluster
+
+import (
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	backingStorageClassLabel = "ocs.openshift.io.backingstorageclass"
+)
+
+type backingStorageClasses struct{}
+
+// ensureCreated ensures that backing storageclasses are created for the StorageCluster
+func (obj *backingStorageClasses) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	platform, err := r.platform.getPlatform(r.Client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	existingBackingStorageClasses, err := getBackingStorageClasses(r)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for i := range sc.Spec.BackingStorageClasses {
+		bsc := &sc.Spec.BackingStorageClasses[i]
+		err := createOrUpdateBackingStorageclass(r, bsc, platform)
+		if err != nil {
+			r.Log.Error(err, "Failed to create StorageClass.", "StorageClass", bsc.Name)
+			continue
+		}
+		findAndDeleteStorageClassFromList(existingBackingStorageClasses, bsc.Name)
```

Comment:
First, this op is poorly named as you don't really delete anything, you remove it from a slice. 
Second, you can use a `map` for a much cleaner approach:
1. After fetching the StoragClassList, you create an index of `map[string]*BackingStorageClass` and fill it.
2. At this location you replace this code with a one-liner   `m[bsc.Name] = nil`
3. You replace the cleanup code (below this loop) with a loop that runs on the map values (not keys) and ignores nils.



The Idea is to create a map[string]bool and initialize it after fetching the storage class list with sc names to false.